framework/src/modules/chain/dpos/account.js

```diff
@@ -0,0 +1,263 @@
+const BigNum = require('@liskhq/bignum');
+
+const _mergeRewardsAndDelegates = (delegatePks, rewards) =>
+	delegatePks.reduce((arr, delegatePk, index) => {
+		let delegate = arr.find(d => d.publicKey === delegatePk);
+
+		if (!delegate) {
+			delegate = {
+				publicKey: delegatePk,
+				reward: new BigNum(0),
+				blockCount: 0,
+			};
+			arr.push(delegate);
+		}
+
+		delegate.reward = delegate.reward.plus(rewards[index]);
+		delegate.blockCount += 1;
```

Comment:
I'm missing some context yet for this method but it's correct to always increment blockCount even though in the previous `if` we add a delegate with zero as blockCount?

Reply:
It's correct. The previous assignment is to initialise the object if not present yet.